src/main/java/ru/l3r8y/prefixed/PrefixedCop.java

```diff
@@ -23,21 +23,130 @@
  */
 package ru.l3r8y.prefixed;
 
+import io.github.classgraph.AnnotationInfo;
+import io.github.classgraph.AnnotationParameterValue;
+import io.github.classgraph.ClassGraph;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ClassInfoList;
+import io.github.classgraph.ScanResult;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
-import org.apache.maven.plugin.MojoFailureException;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import ru.l3r8y.prefixed.annotation.Prefixed;
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 /**
  * Prefixed cop.
+ * This class designed with focus on performance, this is
+ * why code may look ugly to you.
+ *
  * @since 0.0.0
  */
 @Mojo(name = "prefixed", defaultPhase = LifecyclePhase.VERIFY)
 public final class PrefixedCop extends AbstractMojo {
 
+    /**
+     * Thread pool to scan project.
+     */
+    private static final ExecutorService EXECUTORS = Executors.newVirtualThreadPerTaskExecutor();
+
+    /**
+     * The project.
+     */
+    @Parameter(defaultValue = "${project}")
+    private MavenProject project;
+
+    /**
+     * The base package.
+     */
+    @Parameter(property = "basePackage", defaultValue = "${project.groupId}")
+    private String basePackage;
+
+    /**
+     * Fail on error flag.
+     */
+    @SuppressWarnings("PMD.ImmutableField")
+    @Parameter(defaultValue = "true")
+    private boolean failOnError = true;
+
     @Override
-    public void execute() throws MojoExecutionException, MojoFailureException {
-        this.getLog().info("Boom!");
+    public void execute() throws MojoExecutionException {
```

Comment:
@l3r8yJ I suggest to decompose this logic into composable decorators, or at least private methods for now: https://www.yegor256.com/2015/02/26/composable-decorators.html